src/content/blog/astro-490.mdx

```diff
@@ -0,0 +1,186 @@
+---
+title: "Astro 4.9"
+description: "Astro 4.9 is now available! This release includes the long-awaited Container API, stabilized experimental features, and more."
+publishDate: "May 23, 2024"
+authors:
+  - erika
+  - ema
+  - matthew
+  - nate
+  - bjorn
+  - sarah
+coverImage: "/src/content/blog/_images/astro-490/header_astro_4_9.webp"
+socialImage: "/src/content/blog/_images/astro-490/og_image_4_9.webp"
+lang: "en"
+---
+
+import BlogSummary from "/src/components/BlogSummary.astro";
+
+<BlogSummary>
+  Astro 4.9 is out! This release includes the long-awaited Container API, stabilized experimental features, and more. A small but mighty release!
+</BlogSummary>
+
+Full release highlights include:
+
+- [**Experimental: Container API**](#experimental-container-api)
+- [**React 19 support for Astro Actions**](#react-19-support)
+- [**Stabilized experimental features**](#stabilized-experimental-features)
+
+To upgrade an existing project, use the automated `@astrojs/upgrade` CLI tool. Alternatively, upgrade manually by running the upgrade command for your package manager:
+
+```sh
+# Recommended:
+npx @astrojs/upgrade
+
+# Manual:
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+## Experimental: Container API
+
+Astro 4.9 introduces a first look at the long-awaited Container API. This API allows you to render Astro components outside of an Astro application. This is similar to how you can render React components on the server using libraries like `react-dom/server`, or Preact components using `preact-render-to-string`.
+
+Here's an example of how you can use the Container API to render an Astro component to a string:
+
+```ts
+import { experimental_AstroContainer as AstroContainer } from 'astro/container';
+import Component from './components/Component.astro';
+
+const container = await AstroContainer.create();
+console.log(await container.renderToString(Component, { props: {}, slots: {} }))
```

Comment:
Maybe for this initial example of the API it’s worth simplifying it to just show “hey you can render `Component` to a string”:

```suggestion
console.log(await container.renderToString(Component));
```